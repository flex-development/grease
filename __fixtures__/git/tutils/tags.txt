tutils@6.0.0-alpha.24
tutils@6.0.0-alpha.23
tutils@6.0.0-alpha.22
tutils@6.0.0-alpha.21
tutils@6.0.0-alpha.20
tutils@6.0.0-alpha.19
tutils@6.0.0-alpha.18
tutils@6.0.0-alpha.17
tutils@6.0.0-alpha.16
tutils@6.0.0-alpha.15
tutils@6.0.0-alpha.14
tutils@6.0.0-alpha.13
tutils@6.0.0-alpha.12
tutils@6.0.0-alpha.11
tutils@6.0.0-alpha.10
tutils@6.0.0-alpha.9
tutils@6.0.0-alpha.8
tutils@6.0.0-alpha.7
tutils@6.0.0-alpha.6
tutils@6.0.0-alpha.5
tutils@6.0.0-alpha.4
tutils@6.0.0-alpha.3
tutils@6.0.0-alpha.2
tutils@6.0.0-alpha.1
tutils@5.0.1
tutils@5.0.0
tutils@5.0.0-alpha.1
tutils@5.0.0-dev.2
tutils@5.0.0-dev.1
tutils@4.9.0-dev.1
tutils@4.8.0
tutils@4.7.0
tutils@4.6.0
tutils@4.5.0
tutils@4.4.0
tutils@4.3.0
tutils@4.2.3
tutils@4.2.2
tutils@4.2.1
tutils@4.2.0
tutils@4.1.1
tutils@4.1.0
tutils@4.0.3
tutils@4.0.2
tutils@4.0.1
tutils@4.0.0
tutils@4.0.0-dev.0
tutils@3.1.7
tutils@3.1.6
tutils@3.1.5
tutils@3.1.4
tutils@3.1.3
tutils@3.1.2
tutils@3.1.1
tutils@3.1.0
tutils@3.0.0
tutils@2.0.0
tutils@1.0.0
