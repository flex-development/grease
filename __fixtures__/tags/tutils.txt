tag: tutils@6.0.0-alpha.22
















tag: tutils@6.0.0-alpha.21





tag: tutils@6.0.0-alpha.20















tag: tutils@6.0.0-alpha.19















tag: tutils@6.0.0-alpha.18

tag: tutils@6.0.0-alpha.17







tag: tutils@6.0.0-alpha.16



tag: tutils@6.0.0-alpha.15


















tag: tutils@6.0.0-alpha.14













tag: tutils@6.0.0-alpha.13











tag: tutils@6.0.0-alpha.12






tag: tutils@6.0.0-alpha.11



























































































































































































































































tag: tutils@6.0.0-alpha.10
























































































tag: tutils@6.0.0-alpha.9






tag: tutils@6.0.0-alpha.8






























tag: tutils@6.0.0-alpha.7

tag: tutils@6.0.0-alpha.6


tag: tutils@6.0.0-alpha.5





tag: tutils@6.0.0-alpha.4

tag: tutils@6.0.0-alpha.3

tag: tutils@6.0.0-alpha.2




tag: tutils@6.0.0-alpha.1














































tag: tutils@5.0.1










































































tag: tutils@5.0.0




tag: tutils@5.0.0-alpha.1




tag: tutils@5.0.0-dev.2







































tag: tutils@5.0.0-dev.1






tag: tutils@4.9.0-dev.1













































tag: tutils@4.8.0

tag: tutils@4.7.0

tag: tutils@4.6.0





tag: tutils@4.5.0




tag: tutils@4.4.0





tag: tutils@4.3.0

tag: tutils@4.2.3

tag: tutils@4.2.2



tag: tutils@4.2.1








tag: tutils@4.2.0






tag: tutils@4.1.1

tag: tutils@4.1.0












tag: tutils@4.0.3

tag: tutils@4.0.2









tag: tutils@4.0.1







tag: tutils@4.0.0







tag: tutils@4.0.0-dev.0










tag: tutils@3.1.7


tag: tutils@3.1.6


tag: tutils@3.1.5



tag: tutils@3.1.4


tag: tutils@3.1.3


tag: tutils@3.1.2


tag: tutils@3.1.1










































tag: tutils@3.1.0



tag: tutils@3.0.0







tag: tutils@2.0.0






tag: tutils@1.0.0




