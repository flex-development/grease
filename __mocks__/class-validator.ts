import * as item from 'class-validator'

/**
 * @file Node Module Mock - class-validator
 * @module mocks/class-validator
 * @see https://jestjs.io/docs/manual-mocks#mocking-node-modules
 * @see https://github.com/typestack/class-validator
 */

const actual = jest.requireActual<typeof item>('class-validator')

export const buildMessage = jest.fn(actual.buildMessage)
export const validate = jest.fn(actual.validate)
export const validateOrReject = jest.fn(actual.validateOrReject)
export const IsBoolean = jest.fn(actual.IsBoolean)
export const IsEnum = jest.fn(actual.IsEnum)
export const IsNotEmpty = jest.fn(actual.IsNotEmpty)
export const IsOptional = jest.fn(actual.IsOptional)
export const IsString = jest.fn(actual.IsString)
export const ValidateBy = jest.fn(actual.ValidateBy)
export const ValidateIf = jest.fn(actual.ValidateIf)
export const ValidateNested = jest.fn(actual.ValidateNested)
export const ValidatorConstraint = jest.fn(actual.ValidatorConstraint)
