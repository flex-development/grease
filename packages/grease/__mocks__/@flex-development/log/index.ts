/**
 * @file Node Module Mock - @flex-development/log
 * @module grease/tests/mocks/flex-development/log
 * @see https://jestjs.io/docs/next/manual-mocks#mocking-node-modules
 * @see https://github.com/flex-development/log
 */

export default jest.fn((...args) => {
  return jest.requireActual('@flex-development/log').default(...args)
})
