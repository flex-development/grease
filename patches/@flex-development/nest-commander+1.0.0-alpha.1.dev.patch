diff --git a/dist/providers/command-runner.service.mjs b/dist/providers/command-runner.service.mjs
index v1.0.0-alpha.1..v1.0.0-alpha.1 100644
--- a/dist/providers/command-runner.service.mjs
+++ b/dist/providers/command-runner.service.mjs
@@ -15,1 +15,1 @@
-import { cast, constant, define, entries, fallback, flat, get, isString, isUndefined, lowercase, pick, select } from "@flex-development/tutils";
+import { cast, constant, define, entries, fallback, flat, get, isString, isUndefined, lowercase, noop, pick, select } from "@flex-development/tutils";
@@ -76,1 +76,8 @@
-      env && option.env(env), choices.length && option.choices(choices), option.argParser(method.handler.bind(metadata.instance)), command.addOption(option);
+      env && option.env(env);
+      choices.length && option.choices(choices);
+      const { parseArg = noop } = option;
+      option.argParser((value, previous) => {
+        parseArg(value, previous)
+        return method.handler.bind(metadata.instance)(value, previous)
+      });
+      command.addOption(option);
