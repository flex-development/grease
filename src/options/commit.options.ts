/**
 * @file Options - CommitOptions
 * @module grease/options/CommitOptions
 */

import { get } from '@flex-development/tutils'
import { IsArray, IsString } from 'class-validator'
import GitOptions from './git.options'

/**
 * Commit retrieval and parsing options.
 *
 * @class
 * @extends {GitOptions}
 */
class CommitOptions extends GitOptions {
  /**
   * Revision range start.
   *
   * @default ''
   *
   * @public
   * @instance
   * @member {string} from
   */
  @IsString()
  public from: string

  /**
   * Issue reference prefixes.
   *
   * @default ['#','gh-']
   *
   * @public
   * @instance
   * @member {string[]} issue_prefixes
   */
  @IsArray()
  @IsString({ each: true })
  public issue_prefixes: string[]

  /**
   * Revision range end.
   *
   * @default 'HEAD'
   *
   * @public
   * @instance
   * @member {string} to
   */
  @IsString()
  public to: string

  /**
   * Create a new options object.
   *
   * @param {Partial<CommitOptions>?} [opts] - Commit options
   */
  constructor(opts?: Partial<CommitOptions>) {
    super(opts)
    this.from = get(opts, 'from', '')
    this.issue_prefixes = get(opts, 'issue_prefixes', ['#', 'gh-'])
    this.to = get(opts, 'to', 'HEAD')
  }
}

export default CommitOptions
