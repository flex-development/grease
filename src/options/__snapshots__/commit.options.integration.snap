// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`integration:options/CommitOptions > validation > should fail validation if schema is invalid 1`] = `
[
  ValidationError {
    "children": [],
    "constraints": {
      "isString": "from must be a string",
    },
    "property": "from",
    "value": null,
  },
  ValidationError {
    "children": [],
    "constraints": {
      "isArray": "issue_prefixes must be an array",
    },
    "property": "issue_prefixes",
    "value": "#",
  },
  ValidationError {
    "children": [],
    "constraints": {
      "isString": "to must be a string",
    },
    "property": "to",
    "value": /\\^HEAD/,
  },
  ValidationError {
    "children": [],
    "constraints": {
      "isDirectory": "cwd must be an absolute directory path",
    },
    "property": "cwd",
    "value": "file:///",
  },
  ValidationError {
    "children": [],
    "constraints": {
      "isBoolean": "debug must be a boolean value",
    },
    "property": "debug",
    "value": null,
  },
]
`;
